### ISSUE TITLE
Implement TKEO algorithm on iPhone for offline pinch event detection

### ISSUE THREAD

**Issue by moussaba** [moussaba/dhikr-counter#21]
## Summary
Implement the TKEO (Teager-Kaiser Energy Operator) pinch detection algorithm on the iPhone app to analyze received sensor data and generate pinch events for better accuracy and validation.

## Background
Currently, pinch detection happens only on the Apple Watch with a simpler algorithm. The iPhone receives raw sensor data but doesn't perform any analysis. We want to implement the advanced TKEO algorithm on the iPhone to:
- Provide more accurate pinch detection using the full dataset
- Generate ground truth pinch events for algorithm validation  
- Compare Watch vs iPhone detection results
- Enable post-processing analysis of sessions

## Current Architecture
```
Apple Watch → Raw Sensor Data → iPhone (Storage Only)
                                     ↓
                               Analysis Scripts (External)
```

## Proposed Architecture  
```
Apple Watch → Raw Sensor Data → iPhone (TKEO Analysis) → Pinch Events
                                     ↓                        ↓
                               File Storage              UI Display
```

### Core TKEO Implementation
- **Port Algorithm**: Translate Python TKEO algorithm to Swift
- **Multi-Template Support**: Implement additive, multiplicative, and gate-only modes
- **Configuration System**: Use same parameters as Python version
- **Real-Time Processing**: Process data as it's received from Watch

#### Data Processing Pipeline
```swift
class TKEOPinchDetector {
    func processSession(_ sessionData: [SensorReading]) -> [PinchEvent]
    func processRealTime(_ reading: SensorReading) -> PinchEvent?
    func updateConfiguration(_ config: TKEOConfiguration)
}
```

#### Configuration Management
- Import/export YAML configurations used by Python scripts
- UI for parameter adjustment and algorithm tuning
- Preset configurations (aggressive, balanced, conservative)

#### Results Integration
- Store detected pinch events alongside sensor data  
- Display pinch events in DataVisualizationView charts
- Compare Watch vs iPhone detection results
- Export enhanced data with both raw sensor data and iPhone-detected events

#### Analysis Results View
- **Detection Comparison**: Watch events vs iPhone TKEO events  
- **Algorithm Performance**: Precision, recall when actual count available
- **Interactive Timeline**: Toggle between different detection results
- **Export Options**: Include iPhone pinch events in exports

#### Configuration UI
```swift
struct TKEOConfigurationView: View {
    // Parameter sliders for threshold adjustment
    // Template selection (additive/multiplicative/gate)
    // Real-time detection toggle
    // Validation against expected pinch count
}
```

#### Enhanced Pinch Events
```swift
struct PinchEvent {
    let timestamp: Date
    let detectionScore: Double
    let detectionMethod: DetectionMethod  // .watch, .iPhone_tkeo
    let algorithmVersion: String
    let confidence: Double
}

enum DetectionMethod {
    case watch              // Original watch algorithm  
    case iPhone_tkeo        // iPhone TKEO algorithm
    case manual             // Manual correction
}
```

#### Session Results
```swift
struct SessionAnalysisResult {
    let watchPinchEvents: [PinchEvent]
    let iPhonePinchEvents: [PinchEvent]  
    let actualPinchCount: Int?
    let algorithmComparison: AlgorithmComparison
}
```

### Phase 1: Core Algorithm Port
- [ ] Port TKEO energy operator calculation to Swift
- [ ] Implement robust statistics (MAD-based z-scores)
- [ ] Create multi-template detection system
- [ ] Add comprehensive unit tests

### Phase 2: Integration
- [ ] Integrate with PhoneSessionManager data reception
- [ ] Add configuration management system
- [ ] Store iPhone pinch events in session files
- [ ] Update data visualization to show iPhone events

### Phase 3: UI Enhancement
- [ ] Add TKEO configuration screen
- [ ] Enhance DataVisualizationView with algorithm comparison
- [ ] Add detection method filtering and visualization
- [ ] Implement real-time analysis toggle

### Phase 4: Validation
- [ ] Compare iPhone vs Watch detection accuracy
- [ ] Validate against actual pinch counts when available
- [ ] Performance optimization for real-time processing
- [ ] Export enhancement with iPhone detection results

### Performance
- **Efficient Processing**: Use Accelerate framework for vector operations
- **Memory Management**: Process data in chunks to avoid memory spikes
- **Threading**: Background processing to avoid UI blocking

### Accuracy Validation
- **Ground Truth Comparison**: Use actualPinchCount when available
- **Algorithm Metrics**: Calculate precision, recall, F1-score
- **Cross-Validation**: Compare with Python TKEO implementation

### Configuration Sync
- **YAML Compatibility**: Ensure iPhone configs work with Python scripts
- **Parameter Validation**: Validate configuration values and provide defaults
- **Export/Import**: Share configurations between iPhone and analysis scripts

## Acceptance Criteria
- [ ] TKEO algorithm accurately ported to Swift with same results as Python
- [ ] Processes received sensor data and generates pinch events
- [ ] Configuration system compatible with existing YAML configs  
- [ ] UI displays comparison between Watch and iPhone detection results
- [ ] iPhone pinch events included in session data exports
- [ ] Performance suitable for real-time processing
- [ ] Validation against actual pinch counts when available

## Benefits
- **Improved Accuracy**: More sophisticated algorithm than current Watch implementation
- **Better Validation**: Compare different detection methods
- **Research Value**: Generate higher quality training data
- **User Insight**: Show users detailed analysis of their sessions

## Priority
**Medium-High** - Important for research accuracy but not critical for basic app functionality.

## Dependencies
- Accelerate framework for performance
- Existing sensor data infrastructure
- YAML parsing library for configuration

## Labels
`enhancement`, `iPhone`, `algorithm`, `analysis`
